Tasks/BeanstalkDeployApplication/helpers/DeployApplicationTaskOperations.ts

```diff
@@ -163,36 +168,54 @@ export class TaskOperations {
+        await this.sleep(initialStartDelay * 1000);
+
         do {
             tl.debug(`...event poll sleep for ${eventPollDelay}s`);
             await this.sleep(eventPollDelay * 1000);
 
-            const responseEnvironments = await this.beanstalkClient.describeEnvironments(requestEnvironment).promise();
-            if (responseEnvironments.Environments.length === 0) {
-                throw new Error(tl.loc('FailedToFindEnvironment'));
-            }
-            environment = responseEnvironments.Environments[0];
+            // if any throttling exception escapes the sdk's default retry logic,
+            // extend the user's selected poll delay by a random, sub-5 second, amount
+            try {
+                const responseEnvironments = await this.beanstalkClient.describeEnvironments(requestEnvironment).promise();
+                if (responseEnvironments.Environments.length === 0) {
+                    throw new Error(tl.loc('FailedToFindEnvironment'));
+                }
+                environment = responseEnvironments.Environments[0];
 
-            requestEvents.StartTime = lastPrintedEventDate;
-            const responseEvent = await this.beanstalkClient.describeEvents(requestEvents).promise();
+                requestEvents.StartTime = lastPrintedEventDate;
+                const responseEvent = await this.beanstalkClient.describeEvents(requestEvents).promise();
 
-            if (responseEvent.Events.length > 0) {
-                for (let i = responseEvent.Events.length - 1; i >= 0; i--) {
-                    const event = responseEvent.Events[i];
-                    if (event.EventDate <= lastPrintedEventDate) {
-                        continue;
-                    }
+                if (responseEvent.Events.length > 0) {
+                    for (let i = responseEvent.Events.length - 1; i >= 0; i--) {
+                        const event = responseEvent.Events[i];
+                        if (event.EventDate <= lastPrintedEventDate) {
+                            continue;
+                        }
 
-                    console.log(event.EventDate + '   ' + event.Severity + '   ' + event.Message);
+                        console.log(event.EventDate + '   ' + event.Severity + '   ' + event.Message);
 
-                    if (event.Message === 'Failed to deploy application.') {
-                        success = false;
+                        if (event.Message === 'Failed to deploy application.') {
+                            success = false;
+                        }
                     }
-                }
 
-                lastPrintedEventDate = responseEvent.Events[0].EventDate;
+                    lastPrintedEventDate = responseEvent.Events[0].EventDate;
+                }
+            } catch (err) {
+                // if we are still encountering throttles, increase the poll delay some more
+                if (err.code === 'Throttling') {
```

Comment:
I ***highly*** recommend enabling (at minimum) the `no-unsafe-any` tslint rule, which will prevent you from trying to access members of `any`. Without this rule enabled, the type system is almost worthless and type unsafety will propogate throughout the codebase.

Reply:
we have that in the other PR